handbook/debugging.md

```diff
@@ -0,0 +1,71 @@
+# Debugging
+
+One of the most important programming skills to develop is how to figure out what went wrong with your code.
+
+It can be frustrating to write a bunch of code and immediately get a big error, so here's some advice to bear in mind.
+
+## You're not a bad developer
+
+The most important thing is to remember that getting stuck doesn't say anything about you or your skill as a developer.
+
+It's easy for your first reaction to be "I suck at this, no one else has these problems". _Everyone_ makes typos, forgets how things work or gets inscrutable errors in their terminal. The mark of a more experienced developer isn't avoiding errors entirely, it's reacting to them constructively.
+
+## Understanding the problem
+
+Make sure you stop and understand exactly what went wrong before you start trying to fix it. Otherwise you might go down an entirely wrong path, which will make it harder to figure out what the real original problem was later.
+
+### Read the question
```

Comment:
"Read the question" makes sense in some contexts, but not in others. Perhaps just lose this subhead?

Reply:
Yeah that was meant to be metaphorical 😅. As in "read the question" is something my teachers drilled into me at school that is applicable to a bunch of other stuff. Probably not super clear though so dropping it probably makes sense.